### PR TITLE
Image Tools Resource (ITR)

### PR DESCRIPTION
**Pull request by rgozim**
Added ITR page.

See:

 * Feature request: https://trello.com/c/HhPAIrUp
 * Current state: https://rgozim.github.io/idr.openmicroscopy.org/itr.html

v7
screen shot 2017-12-11 at 14 47 49

Major:
 * [x] Move "ITR" link to a submenu of "Analysis"
 * [x] Rework Name/Project fields (Name links to the project URL (clickable); Second column is the Tools reference, pointing to BIII (clickable))
 * [x] Add header text
 * [x] Add datasets json
 * [x] Add itr@ mail alias (Josh)

Main page:
 * [x] Close window option
 * [x] Add project logos (perhaps no longer)
 * [x] Clicking project navigates to the tool website
 * [x] Dataset indicators (R, F, T) on table
 * [x] Legend for dataset indicators
 * [x] BIIS icon next to project icon
 * [x] **GBI logo on all pages**
 * [x] Rename "Datasets" to "Data and Results"
 * [*] Make `Key` horizontal to free up space
 * [ ] **Add `D` (or `I`) for data/images**
 * [ ] Rename tab to "Image Tools Resource"
 * [ ] Clarify `Domain` and `Tag` behavior
 * [ ] Ensure initial name sorting

Popup:
 * [x] **Link to study files / datasets**
 * [x] **Link to top-level (each idr accession number once)**
 * [x] Link "contact author" and "upon request" to `mailto`s.

Data:
 * [ ] add all URLs from biii.eu

Later:
 * [x] Rounded dataset indicators
 * [ ] Filter on click of Dataset Keys
 * [ ] Mouseover on Dataset Keys
 * [ ] Add Orbit as a tool
 * [ ] Let robots search data

### CODE REVIEW

File: _data/tags.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        "code": "AUTOMATED",
+        "url": "http://biii.eu/taxonomy/term/3574"
+    },
+    {
+        "name": "manual",
+        "code": "MANUAL",
+        "url": "http://biii.eu/taxonomy/term/3572"
+    },
+    {
+        "name": "stand alone",
```
Comment (mtbc) [inline _data/tags.json]: Maybe hyphenate "stand-alone"?

**Comment by joshmoore** [comment]
Minor issue running this locally via `bash docker.sh`:

```
[2017-11-29 03:51:45] ERROR `/css/DataTables-1.10.16/images/sort_both.png' not found.
[2017-11-29 03:51:50] ERROR `/css/DataTables-1.10.16/images/sort_asc.png' not found.
[2017-11-29 03:51:53] ERROR `/css/DataTables-1.10.16/images/sort_desc.png' not found.
```

**Comment by rgozim** [comment]
I wouldn't mind hearing any opinions on the look, fee and usability of the ITR page? I'm hoping it meets the requirements for the task.

**Comment by joshmoore** [comment]
I'll try to get time to add some more content here (cc: @eleanorwilliams) especially matching that of the ROIs that we're currently working on, but in terms of UI feedback:

 * I'm unsure about the placement of the "ITR" link the header.
 * Also, "Tools" may end up being a more useful name (or at least the acronym will need explaining somewhere)
 * Show, Search, Paging, and Ordering: :+1:
 * The `Project` field takes up a good deal of space. That may end up being ok, but that likely depends on how substantial the info in the Datasets field is. (forthcoming)
 * If we have a landing page to explain ITR, we'll need a way to navigate back there from the Datatable.
 * Logos is an immediate idea, but I don't know if we want to invest the time.


cc: @lunson @jrswedlow

**Comment by joshmoore** [comment]
Updated the comment above. Do you have an example of what the json for the Dataset field should look like, @rgozim ?

**Comment by joshmoore** [comment]
Talked through potential upsets to the Dataset field with @eleanorwilliams:

 * It's possible that we won't have values for all the rows. So we might want to add text to the top (or the landing page) along the lines of: "These are tools suggested for re-analyze of the IDR datasets..."
 * "Datasets" could then become: "Examples of analysis results" (or similar) so that some fields can remain empty.

Assuming we want to link ITR from this version in the menu, perhaps move to the right side:
    - Submissions
    - API
    - Analysis Enviroment (i.e. Jupyter)
    - Analysis Tools
    - Deployment

**Comment by lunson** [comment]
Anyone have content for [ ] below? I'm working on styling so I can throw a description in there while I'm at it:

Image Tools Resource (ITR) is a [insert brief description here]. Below you will find [brief explanation of what table summarizes].

**Comment by joshmoore** [comment]
@jasonswedlow: this is now live at https://joshmoore.github.io/idr.openmicroscopy.org/ for review

@lunson: I'll try to write up a blurb here shortly.

@rgozim: after discussing with JRS, a suggestion for the menus:

```
Left:
 - About

Right:
 - Submission (no change to submenu)
 - Analysis
    - API Access
    - Virtual Analysis Environment (VAE)
    - Image Tools Resource (ITR) 
 - Deployment
```

cc: @manics @eleanorwilliams

**Comment by manics** [comment]
idr-fullpage
- URLs aren't clickable
- Entries under `Domains` look like they should be tags that can be clicked to filter results
- Same goes for `Tags`, I'd expect to click on a tag and see all items with that tag instead of being taken to a http://biii.eu/taxonomy page
- Are the names in any particular order?
- Tags would look better if there was something to separate them even though they're on separate lines. E.g. a border like with omero.web tags, or a separator such as `,` or `|`.

**Comment by lunson** [comment]
@rgozim Here's my branch with styles for the page and the table: https://github.com/lunson/idr-demo.openmicroscopy.org/tree/style-itr-page I will push copy for the hero section after I get it from @joshmoore. Thanks!

**Comment by joshmoore** [comment]
@jasonrswedlow also mentioned having potentially non-IDR datasets listed under "Datasets".

@lunson: https://joshmoore.github.io/idr.openmicroscopy.org/itr.html updated with your CSS

NB: I've collected the items here into a checklist in the description of this PR.

**Comment by rgozim** [comment]
@joshmoore I was looking at using project icons alongside the project name, however there is not a lot of consistency in them. A number of the images will need to be edited and have accompanying text removed. Anyone good with photoshop?

**Comment by joshmoore** [comment]
@rgozim: safe to de-prioritize the logos for the moment (esp. if costly)

**Comment by joshmoore** [comment]
Updated text at https://joshmoore.github.io/idr.openmicroscopy.org/itr.html

screen shot 2017-12-01 at 08 34 57

**Comment by eleanorwilliams** [comment]
@joshmoore I think we need more explanation of what the tags are.  Some take you to the EDAM ontology (e.g. feature extraction) but others take you to http://biii.eu/taxonomy/ (e.g. 'manual - http://biii.eu/taxonomy/term/3572) with no explantation of what they mean.  Should we mix the two kind of links, or just have one or the other type?

Maybe the blurb above could say 'The tags describe what the software is used for. Clicking on the tag will take you to ..... To filter software by the tags enter the tag term in the search box. '

**Comment by joshmoore** [comment]
My preference would have been that all the terms were in one, but for the mock values I was adding to the presentation that wasn't the case. It might make more sense to have a strict `EDAM` column to go along with the strict `BIII` column.

**Comment by joshmoore** [comment]
v3:

screen shot 2017-12-06 at 11 37 46

screen shot 2017-12-06 at 11 37 50


Feedback:
 * concerned about logos (JRS)
 * need some form of overview column of how many datasets, etc. (EW)
 * (TBD)

**Comment by rgozim** [comment]
Latest changes include:
- Clicking project navigates to the tool website
- Dataset present indicators (R, F, T) on table
- Legend to left of table displaying legend for dataset indicators
- Clicking biis icon navigates to biis website

@lunson perhaps you could help with some styling, if you have time?

**Comment by joshmoore** [comment]
@lunson : latest commit is https://github.com/rgozim/idr.openmicroscopy.org/commit/78d84bb92714951a22344588c2550eb801df37c5

**Comment by lunson** [comment]
Sorry, I'm having trouble getting what is currently previewed on https://rgozim.github.io/idr.openmicroscopy.org/itr.html onto a local branch for me to make changes so here's what I'd do style-wise:

- Replace the blurry sort arrows graphics with http://fontawesome.io/icon/caret-up/ and http://fontawesome.io/icon/caret-down/ - you only need to show one caret at a time (and FontAwesome is already included in the repo).
- I don't think the logos add much value and find them distracting since they're of varying qualities, but I realize that's probably just me so if we're going to keep them: 
   1. Use the OMERO logo (not OME logo) for the OMERO entry
   2. Move the bii.edu logo (and better yet, replace it with http://fontawesome.io/icon/external-link/) to the right of the Name so as not to compete with the project logos, and to allow for all the Names to left-align with each other.
- Switch cursor over Datasets to default or pointer to better indicate that they are clickable.
- Datasets should also be circular if the Tags Key shows them as circles.
- Are values for the Tags Key expected to grow? If it's going to just be 3-4 Tags, then I'd move them along the top across a row below the Show and Search, to give the table more horizontal real estate. Otherwise, leave as is.

**Comment by joshmoore** [comment]
v5 now migrated to https://joshmoore.github.io/idr.openmicroscopy.org

screen shot 2017-12-07 at 14 56 55

File: itr.html [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        </div>
+    </div>
+</header>
+
+<hr class="invisible">
+<div class="row">
+    <div class="small-12 small-centered medium-12 medium-centered columns">
+        <div class="row horizontal">
+            <div>
+                <p>Below you will find community driven
```
Comment (mtbc) [inline itr.html]: hyphenate community-driven

Comment (mtbc) [inline itr.html]: Maybe have a bit of a left/right margin for this text, for me it goes right to the edge.

File: itr.html [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                    <span class="bold">projects</span>, their entry in the Bio Image Information Index
+                    (<a href="http://biii.eu">http://biii.eu</a>),
+                    <span class="bold">domains</span> in which the software is applicable,
+                    <span class="bold">tags</span> sampling the tasks that the software is useful for, and a list of
+                    <span class="bold">datasets</span>, both in the IDR and elsewhere.</p>
+                <p>If you’d like to submit tools to the ITR, please email us at
+                    <a href="mailto:idr@openmicroscopy.org">itr@openmicroscopy.org</a>.</p>
```
Comment (mtbc) [inline itr.html]: Text doesn't match link.

File: itr.html [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        </div>
+    </div>
+</div>
+
+<div class="row itrBody">
+    <!-- Left column displaying keys -->
+    <div class="small-3 columns">
+        <ul class="accordion" data-accordion data-allow-all-closed="true" data-multi-expand="true">
+            <li class="accordion-item is-active" data-accordion-item>
+                <a href="#" class="accordion-title">Tags Key</a>
```
Comment (mtbc) [inline itr.html]: These seem to appear in the Datasets column rather than the Tags column.

Comment (joshmoore) [reply itr.html]: Good point. Let me see if I can come up with a name.

**Comment by lunson** [comment]
https://github.com/lunson/idr-demo.openmicroscopy.org/tree/style-itr-table (based off of @rgozim's `itr` branch) for latest styles. Let me know if you prefer I open a PR against @rgozim.

**Comment by joshmoore** [comment]
All good, @lunson: I've merged here. https://joshmoore.github.io/idr.openmicroscopy.org/itr.html updated.

**Comment by joshmoore** [comment]
Thanks, @eleanorwilliams

**Comment by joshmoore** [comment]
v6 still on https://joshmoore.github.io/idr.openmicroscopy.org/itr.html with new styling from @lunson  as well as links to studies and authors

screen shot 2017-12-07 at 17 42 42

screen shot 2017-12-07 at 17 42 50

**Comment by joshmoore** [comment]
@lunson / @rgozim : generally big 👍 . Think the biggest remaining request is a horizontal key to free up space width-wise.

**Comment by joshmoore** [comment]
At the moment, we don't have anything more than 4. I don't think it will ever go much higher, but certainly not at the moment.

**Comment by lunson** [comment]
Pushed a style for a horizontal key in https://github.com/lunson/idr-demo.openmicroscopy.org/tree/style-itr-table, but need someone more script-savvy than I to place it between the Show/Search row and the table.

**Comment by joshmoore** [comment]
The styling didn't come out right, but I've placed the keys below the Show/Search:
screen shot 2017-12-11 at 14 13 56

**Comment by joshmoore** [comment]
v7 with horizontal keys now live

screen shot 2017-12-11 at 14 47 49

**Comment by joshmoore** [comment]
Deliverable document has been submitted! Merging & deplying to idr-testing.

Note: there may be a javascript issue on some less-used browsers which will be fixed up separately.

**Comment by joshmoore** [comment]
LGTM. Thanks, @rgozim

**Pull request merged**